src/language-server/analyzer-synchronizer.ts

```diff
@@ -114,19 +115,38 @@ export default class AnalyzerSynchronizer extends AutoDisposable {
  */
 export class LsAnalyzer extends Analyzer {
   constructor(
-      private documents: TextDocuments,
+      private documents: TextDocuments, private logger: Logger,
       private analysisVersionMap:
           WeakMap<Analysis, ReadonlyMap<string, number>>,
       options: AnalyzerOptions) {
     super(options);
   }
 
-  analyze(files: string[]): Promise<Analysis> {
-    return this.annotateWithVersionMap(super.analyze(files));
+  analyze(files: string[], reason?: string): Promise<Analysis> {
+    const prefix = reason ? `${reason}: ` : '';
+    this.logger.log(`${prefix}Analyzing files: ${files.join(', ')}`);
+    const start = +new Date();
```

Comment:
nit: consider using `new Date().getTime()`, it's faster & clearer when reading
https://jsperf.com/date-vs-get-time
https://stackoverflow.com/questions/14255664/new-date-is-this-good-practice

Reply:
+1, done